umh-core/pkg/service/redpanda/redpanda.go

```diff
@@ -0,0 +1,813 @@
+	StartRedpanda(ctx context.Context) error
+	// StopRedpanda stops a Redpanda instance
+	StopRedpanda(ctx context.Context) error
+	// ForceRemoveRedpanda removes a Redpanda instance from the S6 manager
+	ForceRemoveRedpanda(ctx context.Context) error
+	// ServiceExists checks if a Redpanda service exists
+	ServiceExists(ctx context.Context) bool
+	ReconcileManager(ctx context.Context, tick uint64) (error, bool)
+	// IsLogsFine checks if the logs of a Redpanda service are fine
+	// Expects logs ([]s6service.LogEntry), currentTime (time.Time), and logWindow (time.Duration)
+	IsLogsFine(logs []s6service.LogEntry, currentTime time.Time, logWindow time.Duration) bool
+	// IsMetricsErrorFree checks if the metrics of a Redpanda service are error-free
+	IsMetricsErrorFree(metrics Metrics) bool
+	// HasProcessingActivity checks if a Redpanda service has processing activity
+	HasProcessingActivity(status RedpandaStatus) bool
+}
+
+// HTTPClient interface for making HTTP requests
+type HTTPClient interface {
+	Do(req *http.Request) (*http.Response, error)
+}
+
+// defaultHTTPClient is the default implementation of HTTPClient
+type defaultHTTPClient struct {
+	client *http.Client
+}
+
+func newDefaultHTTPClient() *defaultHTTPClient {
+	transport := &http.Transport{
+		MaxIdleConns:      10,
+		IdleConnTimeout:   30 * time.Second,
+		DisableKeepAlives: false,
+	}
+
+	return &defaultHTTPClient{
+		client: &http.Client{
+			Transport: transport,
+		},
+	}
+}
+
+func (c *defaultHTTPClient) Do(req *http.Request) (*http.Response, error) {
+	return c.client.Do(req)
+}
+
+// ServiceInfo contains information about a Redpanda service
+type ServiceInfo struct {
+	// S6ObservedState contains information about the S6 service
+	S6ObservedState s6fsm.S6ObservedState
+	// S6FSMState contains the current state of the S6 FSM
+	S6FSMState string
+	// RedpandaStatus contains information about the status of the Redpanda service
+	RedpandaStatus RedpandaStatus
+}
+
+// RedpandaStatus contains information about the status of the Redpanda service
+type RedpandaStatus struct {
+	// HealthCheck contains information about the health of the Redpanda service
+	HealthCheck HealthCheck
+	// Metrics contains information about the metrics of the Redpanda service
+	Metrics Metrics
+	// MetricsState contains information about the metrics of the Redpanda service
+	MetricsState *RedpandaMetricsState
+	// Logs contains the logs of the Redpanda service
+	Logs []s6service.LogEntry
+}
+
+// HealthCheck contains information about the health of the Redpanda service
+// https://docs.redpanda.com/redpanda-connect/guides/monitoring/
+type HealthCheck struct {
+	// IsLive is true if the Redpanda service is live
+	IsLive bool
+	// IsReady is true if the Redpanda service is ready to process data
+	IsReady bool
+	// Version contains the version of the Redpanda service
+	Version string
+}
+
+// Metrics contains information about the metrics of the Redpanda service
+type Metrics struct {
```

Comment:
potentially also a list of all the topics and their partitions. could be maybe interesting in the future to have it and show it in the console